### PR TITLE
[1132] Added a dedicated auth for backend

### PR DESCRIPTION
**Pull request by defong**
### Context
A dedicated auth for backend as it shouldn't be using the existing `BearTokenApiKey*`.

### Changes proposed in this pull request
Added a dedicated auth for backend inclusive with the using the new env key ` manage_courses_backend:key`

### Guidance to review
This is similar to approach as `BearTokenApiKey*` and uses the same handler and options, only real changes is just the niceties.

### CODE REVIEW

**Comment by defong** [comment]
Related to https://github.com/DFE-Digital/bat-infrastructure/pull/119

**Comment by defong** [comment]
Related to https://github.com/DFE-Digital/manage-courses-api/pull/281

**Comment by tvararu** [comment]
[Config in Manage Courses Backend for bearer token auth with Manage Courses API](https://trello.com/c/i9USTN0k/1133-config-in-manage-courses-backend-for-bearer-token-auth-with-manage-courses-api)

File: src/ManageCourses.Api/appsettings.Development.json [hunk 1]
```diff
@@ -8,10 +8,11 @@
...
   },
-  "PG_USERNAME":"manage_courses",
-  "PG_PASSWORD":"manage_courses",
-  "auth:oidc:userinfo_endpoint":"https://signin-test-oidc-as.azurewebsites.net/me",
-  "api:key":"chicken",
+  "PG_USERNAME": "manage_courses",
+  "PG_PASSWORD": "manage_courses",
+  "auth:oidc:userinfo_endpoint": "https://signin-test-oidc-as.azurewebsites.net/me",
+  "api:key": "chicken",
+  "manage_courses_backend:key": "rubbish-ruby",
```
Comment (tvararu) [inline src/ManageCourses.Api/appsettings.Development.json]: Can you change this to `bananas` please?

Comment (defong) [reply src/ManageCourses.Api/appsettings.Development.json]: Sure fat `Banana` it be than.

**Comment by defong** [comment]
[Manage Courses API endpoint for triggering course-republishing](https://trello.com/c/ooKtcota/1132-manage-courses-api-endpoint-for-triggering-course-republishing)

File: src/ManageCourses.Api/McConfig.cs [hunk 2]
```diff
@@ -12,6 +12,7 @@ public class McConfig
         private const string ConfigKeyForPgServer = "MANAGE_COURSES_POSTGRESQL_SERVICE_HOST";
         private const string ConfigKeyForSignInUserInfoEndpoint = "auth:oidc:userinfo_endpoint";
         private const string ConfigKeyForApiKey = "api:key";
+        private const string ConfigKeyForBackendApiKey = "manage_courses_backend:key";
```
Comment (timabell) [inline src/ManageCourses.Api/McConfig.cs]: with hindsight I'm not that keen on these nested things, what do you think?

File: src/ManageCourses.Api/appsettings.Development.json [hunk 3]
```diff
@@ -8,10 +8,11 @@
...
   },
-  "PG_USERNAME":"manage_courses",
-  "PG_PASSWORD":"manage_courses",
-  "auth:oidc:userinfo_endpoint":"https://signin-test-oidc-as.azurewebsites.net/me",
-  "api:key":"chicken",
+  "PG_USERNAME": "manage_courses",
+  "PG_PASSWORD": "manage_courses",
+  "auth:oidc:userinfo_endpoint": "https://signin-test-oidc-as.azurewebsites.net/me",
+  "api:key": "chicken",
+  "manage_courses_backend:key": "banana",
```
Comment (timabell) [inline src/ManageCourses.Api/appsettings.Development.json]: :+1: :banana:

File: src/ManageCourses.Api/McConfig.cs (hunk 2 above)
Comment (defong) [reply src/ManageCourses.Api/McConfig.cs]: Should of went with the Options pattern

**Pull request merged**